File: algorithms_toolbox/week2_algorithmic_warmup/2_last_digit_of_fibonacci_number/fibonacci_last_digit.cpp
#include <iostream>

long long get_fibonacci_last_digit_naive(long long n) {
    if (n <= 1)
        return n;

    long long previous = 0;
    long long current  = 1;
    long long result = 0;

    for (long long i = 0; i < n - 1; ++i) {
        result = (previous + current) % 10;
        long long tmp_previous = previous;
        previous = current;
        current = tmp_previous + current;

    }

    return result;
}

int main() {
    long long n;
    std::cin >> n;
    long long c = get_fibonacci_last_digit_naive(n);
    std::cout << c << '\n';
    }

File: coursera/algorithms_toolbox_problems/week2_algorithmic_warmup/2_last_digit_of_fibonacci_number/fibonacci_last_digit.py
# Uses python3
import sys


def get_fibonacci_last_digit_naive(n):
    if n <= 1:
        return n
    result = 0
    previous = 0
    current  = 1

    for _ in range(n - 1):
        result = (previous + current) % 10
        previous, current = current, result


    return result


n = int(input())
print(get_fibonacci_last_digit_naive(n))
